### PR TITLE
[Bugfix] Throw exception when failed to get databases

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/7705

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 1]
```diff
@@ -461,7 +461,7 @@ private void handleShowProc() throws AnalysisException {
     }
 
     // Show databases statement
-    private void handleShowDb() throws AnalysisException {
+    private void handleShowDb() throws AnalysisException, DdlException {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:464]: I will normalize exception in a few days

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|1|2|50.00%|

**Review by gengjun-git — APPROVED** [review]

**Comment by stephen-shelby** [comment]
@Mergifyio backport branch-2.3

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|1|2|50.00%|

**Pull request merged**